Added the route function

ezimuel opened the pull request:
This PR adds a new function `route()` in the view. This function returns a `Zend\Expressive\Router\RouteResult` object in a view. For instance, this can be useful to retrieve the matched route name in a view (or template) using the command `$this->route()->getMatchedRouteName()` or the matched parameters using `$this->route()->getMatchedParams()`.

**NOTE:** This PR needs https://github.com/zendframework/zend-expressive-helpers/pull/71 to be merged.

ezimuel committed:
Added the route function

michalbundyra commented on src/Exception/RuntimeException.php:
```diff
@@ -0,0 +1,14 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-platesrenderer for the canonical source repository
+ * @copyright Copyright (c) 2016-2017 Zend Technologies USA Inc. (https://www.zend.com)
```
2019 only, please

michalbundyra commented on src/Extension/UrlExtension.php:
```diff
@@ -48,6 +50,22 @@ public function register(Engine $engine) : void
+    /**
+     * Get the RouteResult instance of UrlHelper
+     *
+     * @return RouteResult
```
it is useless at is not gives more information than function declaration (RTH)

michalbundyra commented on src/Extension/UrlExtension.php:
```diff
@@ -48,6 +50,22 @@ public function register(Engine $engine) : void
+            throw new Exception\RuntimeException(
+                'The helper does not contains the route result. Please update zend-expressive-helpers'
+            );
+        }
```
do we need that exception? Maybe we should bump minimum version of `zend-expressive-helpers` once released?

If needed, we should add `@throws` annotation on the method.

weierophinney replied:
I've bumped the minimum supported zend-expressive-helpers version to 5.2 at this time, which allows removing the new exception, as well as the check here.

weierophinney committed:
refactor: depend on zend-expressive-helpers 5.2.0

Doing so ensures that the `UrlHelper::getRouteResult()` method is
publicly visible, which in turn obviates the need for having the new
`RuntimeException`.

weierophinney committed:
fix: make getRouteResult() have a nullable return value

This is required as `UrlHelper::getRouteResult()` has a nullable return
value.

michalbundyra commented on test/Extension/UrlExtensionTest.php:
```diff
@@ -12,9 +12,11 @@
 use League\Plates\Engine;
 use PHPUnit\Framework\TestCase;
 use Prophecy\Prophecy\ProphecyInterface;
+use ReflectionMethod;
```
It seems to be not used

weierophinney committed:
qa: remove unused import, update license year

weierophinney committed:
Adds CHANGELOG entry for #36
